### ISSUE TITLE
Integration stops working on March 31, 2025

### ISSUE THREAD

**Issue by Daanoz** [Daanoz/ha-google-photos#64]
Due to Google Photo [API changes](https://developers.googleblog.com/en/google-photos-picker-api-launch-and-library-api-updates/) announced by Google, this integration will stop working on March 31, 2025. The new API will only allow for the integration to access photo's and albums which are uploaded by the integration itself, which renders this integration useless in my opinion.

Alternatively, if you want Home Assistant to upload photos, and download them back again, starting from HA 2024.10, you can use the official [Google Photo's integration](https://rc.home-assistant.io/integrations/google_photos).

**Comment by arcsur** [comment]
@Daanoz What about the picker API?  Sure it is inconvenient making the user choose which photos to allow an app to see via the Google Photos App (URI/QR Code) but it still seems like selection of photos to "share" with an application would be feasible based on the information on that google announcement:

> Migrate to the Picker API: If your app currently uses the Library API for photo selection from content not uploaded by your app, you'll need to [migrate to the new Picker API](https://developers.google.com/photos/picker/guides/get-started-picker) to ensure a seamless user experience and enhanced security.

**Comment by Daanoz** [comment]
Yeah, as I understand it, it's like a one time access thing. So once the user selects "a couple" of photos, the integration can download/store them locally. To me that eliminates a lot of use cases as you would constantly need to update the photos by manually selecting them via HA. If you want something like that you could also just manually download the photos and place them in HA.

However, if you think it makes sense, you could see if the core integration is willing to take this on as a feature.

**Comment by ingoau** [comment]
This could be transitioned into showing photos from Google Drive if that isn't being removed as well...

**Comment by cannondale0815** [comment]
FYI -- the new [official Google Photos integration](https://www.home-assistant.io/integrations/google_photos/) that is available in 2024.10.0 (out now) cannot be configured until one removes this custom integration here (presumably due to exact same name). 

Also, the official integration is only capable of uploading images -- sadly there is no way to manually grant it access to anything else in your photos library.

**Comment by AlenHodnik** [comment]
Your integration was one of the best. Hopefully I don't ask a stupid question, but would it be possible to change this integration without too much hassle to do the same as it did with google photo now with local folder?

**Comment by ingoau** [comment]
> Your integration was one of the best. Hopefully I don't ask a stupid question, but would it be possible to change this integration without too much hassle to do the same as it did with google photo now with local folder? 

Or a Google drive folder for the same functionality

**Comment by Daanoz** [comment]
> Your integration was one of the best. Hopefully I don't ask a stupid question, but would it be possible to change this integration without too much hassle to do the same as it did with google photo now with local folder? 

It's not a bad idea, and I think it should be possible.  It will take some effort though for which I can't estimate when I have the time.

**Comment by trevornorcross** [comment]
> FYI -- the new [official Google Photos integration](https://www.home-assistant.io/integrations/google_photos/) that is available in 2024.10.0 (out now) cannot be configured until one removes this custom integration here (presumably due to exact same name).

This is a problem right now - I can't even test out the new native integration (nobody can who has HACS installed) because the name on the custom integration is the same and overrides the native integration. Right now there is no way for me to even start to setup the native integration without removing HACS and all the integrations I use from that unless custom or native integration get renamed.

**Comment by Daanoz** [comment]
> > FYI -- the new [official Google Photos integration](https://www.home-assistant.io/integrations/google_photos/) that is available in 2024.10.0 (out now) cannot be configured until one removes this custom integration here (presumably due to exact same name).
> 
> This is a problem right now - I can't even test out the new native integration (nobody can who has HACS installed) because the name on the custom integration is the same and overrides the native integration. Right now there is no way for me to even start to setup the native integration without removing HACS and all the integrations I use from that unless custom or native integration get renamed.
> 
> 

Yeah it's quite unfortunate, but not much can be done... I can rename the custom integration, but then you (and everyone else who updates) loses all the current entities and settings. I don't think there is a way to do clean rename integration migration.

If you really want to run them side by side (and reconfigure this integration from scratch) you can rename the "google_photos" directory in the "custom_components" directory. 
Take the new name, and update it in the following places:
- [const.py, line 4](https://github.com/Daanoz/ha-google-photos/blob/7ca34c99943d4f7d7e06da0dfe8a8cb5c4f2edfb/custom_components/google_photos/const.py#L4)
- [\_\_init.py__, line 19](https://github.com/Daanoz/ha-google-photos/blob/7ca34c99943d4f7d7e06da0dfe8a8cb5c4f2edfb/custom_components/google_photos/__init__.py#L19)
- [manifest.json, line 2](https://github.com/Daanoz/ha-google-photos/blob/7ca34c99943d4f7d7e06da0dfe8a8cb5c4f2edfb/custom_components/google_photos/manifest.json#L2)
- [services.yaml, line 6](https://github.com/Daanoz/ha-google-photos/blob/7ca34c99943d4f7d7e06da0dfe8a8cb5c4f2edfb/custom_components/google_photos/services.yaml#L6)

Then restart HA.

**Comment by ingoau** [comment]
> Your integration was one of the best. Hopefully I don't ask a stupid question, but would it be possible to change this integration without too much hassle to do the same as it did with google photo now with local folder?

I might fork or start another project with this at some point...

**Comment by Daanoz** [comment]
> > Your integration was one of the best. Hopefully I don't ask a stupid question, but would it be possible to change this integration without too much hassle to do the same as it did with google photo now with local folder?
> 
> I might fork or start another project with this at some point...

That's probably the fastest at this point, I have a bit of shortage in time recently. Happy to help if you have any questions, you can find me in the HA discord channel.

**Comment by splitsec2** [comment]
> Due to Google Photo [API changes](https://developers.googleblog.com/en/google-photos-picker-api-launch-and-library-api-updates/) announced by Google, this integration will stop working on March 31, 2025. The new API will only allow for the integration to access photo's and albums which are uploaded by the integration itself, which renders this integration useless in my opinion.
> 
> Alternatively, if you want Home Assistant to upload photos, and download them back again, starting from HA 2024.10, you can use the official [Google Photo's integration](https://rc.home-assistant.io/integrations/google_photos).

Would setting up a new google account for use by the plug in and then do partner sharing with it get around this?  Ie my_ha_photo_acct@gmail.com is tied to the add in, and then myaccount@gmail.com does a partner share with them?

**Comment by ingoau** [comment]
> > Due to Google Photo [API changes](https://developers.googleblog.com/en/google-photos-picker-api-launch-and-library-api-updates/) announced by Google, this integration will stop working on March 31, 2025. The new API will only allow for the integration to access photo's and albums which are uploaded by the integration itself, which renders this integration useless in my opinion.
> > 
> > Alternatively, if you want Home Assistant to upload photos, and download them back again, starting from HA 2024.10, you can use the official [Google Photo's integration](https://rc.home-assistant.io/integrations/google_photos).
> 
> Would setting up a new google account for use by the plug in and then do partner sharing with it get around this?  Ie my_ha_photo_acct@gmail.com is tied to the add in, and then myaccount@gmail.com does a partner share with them?

Don't think so as I think the new API transitions to using the picker API, which allows users to be directed to google photos to pick what the website or app (integrating with google photos) can access.

**Comment by boojew** [comment]
The picker api doesn’t force download. It grants access to the items in question - granted out of band or through google ui. It’s similar to how iOS allows you to select photos to be available to an app. 

Speaking of which.. we need an iCloud Photos integration

**Comment by ingoau** [comment]
> Speaking of which.. we need an iCloud Photos integration

Doesn't look like it's possible:

![Image](https://github.com/user-attachments/assets/ea2e56c5-2adc-459d-97d6-dae7b3a99e7a)

https://forums.developer.apple.com/forums/thread/74434

**Comment by MrAct1on** [comment]
Can someone help me figure out how to clear this error? I removed all traces of the integration way back when the announcement was posted. However, I can’t get the repair issue to go away. I’ve tried reinstalling Spook (the integration that reports it) and that didn’t help either. I’ve checked my logs quite a few times and nothing seems to correlate to why it’s still popping up

![Image](https://github.com/user-attachments/assets/fc568e23-46f2-418d-a733-e400a747cbcb)
![Image](https://github.com/user-attachments/assets/f0a35038-515a-4165-88a8-c791bd783dbf)

**Comment by tajnymag** [comment]
I've wanted to use this integration to create a smart photo frame. In such context, going through the new Picker workflow would actually make some sense, IMHO. Still better than having absolutely no Google Photos library integration at all.

Can the Picker API pick whole albums? If it could, the UX would be manageable. I haven't found an answer in Googles announcements.

**Comment by timlaing** [comment]
> > Speaking of which.. we need an iCloud Photos integration
> 
> Doesn't look like it's possible:
> 
> ![Image](https://github.com/user-attachments/assets/ea2e56c5-2adc-459d-97d6-dae7b3a99e7a)
> 
> https://forums.developer.apple.com/forums/thread/74434

FWIW: I am working on an iCloud version of this integration based on this one. Been focusing on the `pyicloud` library updates that will support this.

**Comment by trevornorcross** [comment]
> > > Speaking of which.. we need an iCloud Photos integration
> > 
> > 
> > Doesn't look like it's possible:
> > ![Image](https://github.com/user-attachments/assets/ea2e56c5-2adc-459d-97d6-dae7b3a99e7a)
> > https://forums.developer.apple.com/forums/thread/74434
> 
> FWIW: I am working on an iCloud version of this integration based on this one. Been focusing on the `pyicloud` library updates that will support this.

Any updates on timing or anything, I would sure love to test/try this.

**Comment by timlaing** [comment]
Probably in the next month as I have to get pyicloud to be update in home assistant first. 

I have the first draft of the integration is working so will publish as soon as I can.

**Comment by Asdfuiop** [comment]
The integration still somehow works. How is this possible?

**Comment by Daanoz** [comment]
> The integration still somehow works. How is this possible?

My guess is that it's due to that the "scopes" being used by the integration are no longer supported. Meaning, after the first restart of HomeAssistant, it tries to re-authenticate and then it stops working.

**Comment by Dan-Wilder** [comment]
> > > Speaking of which.. we need an iCloud Photos integration
> > 
> > Doesn't look like it's possible:
> > 
> > ![Image](https://github.com/user-attachments/assets/ea2e56c5-2adc-459d-97d6-dae7b3a99e7a)
> > 
> > https://forums.developer.apple.com/forums/thread/74434
> 
> FWIW: I am working on an iCloud version of this integration based on this one. Been focusing on the `pyicloud` library updates that will support this. 

You're likely aware, but the Dakboard folks have managed to do this with "public" urls for shared albums. I tried the pyicloud method but hit a wall. I was going to do some digging on how Dakboard managed it but ran out of steam.(I think they mainly just figured out the url schema on the cdn. Also happy to test!)

**Comment by Daanoz** [comment]
I'll be archiving the repository for now as I don't want to give the impression that it can be used. This means it will become read-only. If someone has a good alternative ready to go, I'm willing to add it to this thread to point others to an alternative solution. I can be found on Discord on the regular HomeAssistant and HACS channels.
